databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
+      // Type of corresponding Books entity
+      type: 'string',
+      required: false,
+      mustNotBeEmpty: true
+    },
+    kashooId: {
+      // The ID of the corresponding Books entity
+      type: 'integer',
+      required: false,
+      minimumValue: 1
+    },
+    data: {
+      // Raw data of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    }
```

Comment:
In your example, I think the details would not be surfaced here. I imagined that the ML algorithm would pull all `bank` type records that do not have processed = true. In this example, it would create its own intermediate record with internal state and would mark the shoeboxItem as processed (because it already has relevant processing data, whatever it is).

Of course the fact that the doc is processed is also implied by the processing data document existing, so this goes back to this boolean being either used to optimize queries or perhaps for programmer ease-of-use when looking at the docs.